### PR TITLE
Feature/person student and teacher classes

### PR DESCRIPTION
**Pull request by MarcoOquendoC**
# [OOP school library: add Person, Student and Teacher classes](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md)

## Hi good reviewer,

Patricio greeting

## These are the project requirements for this milestone:

- Create class Person.
- Create class Student.
- Create class Teacher.

### CODE REVIEW

File: teacher.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  attr_accessor :specialization
+
+  def initialize(
+    specialization,
+    age,
+    name = 'Unknown'
+  )
+    super(age, name, true)
+    @specialization = specialization
+  end
```
Comment (HAMMAS-SALEEM) [inline teacher.rb]: - [x] Kindly add an paratmeter parent_permission: true in your contstructor and pass it to the parent class 🤔 I know it sounds confusing but you should follow the requirements 👍

**Your code will look something similar to the below ⬇️ code**

```ruby
def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,113 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- CONTRIBUTING -->
+
+## 🤝 Contributing <a name="contributing"></a>
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](../../issues/).
+
```
Comment (HAMMAS-SALEEM) [inline README.md]: - [x] Your link to the issues page is broken 😨 Kindly fix it to make your readme file descriptive 👍

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @MarcoOquendoC :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: teacher.rb (hunk 1 above)
Comment (MarcoOquendoC) [reply teacher.rb]: Hi @HAMMAS-SALEEM 

Are you sure about your requirement?
The teacher does not need the parent permission parameter to be specified and I sent it to the parent class using super to be always true.

**Review by FrederickMih — APPROVED** [review]
## Hi 🙋 @MarcoOquendoC, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlight 🟢 
- PR is professional 👍 
- All Linter checks passed ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**